client/network/src/service.rs

```diff
@@ -812,6 +885,13 @@ impl<B, H> NetworkStateInfo for NetworkService<B, H>
 	}
 }
 
+/// Error returned by [`NetworkService::send_notification`].
+#[derive(Debug, derive_more::Display, derive_more::Error)]
+pub enum SendNotificationError {
+	/// No open notifications substream exists with this combination of peer and protocol.
```

Comment:
Would be easier for me to understand. Feel free to ignore.

```suggestion
	/// No open notifications substream exists with the provided combination of peer and protocol.
```